db/revision_cache_lru.go

```diff
@@ -135,6 +149,35 @@ func (rc *LRURevisionCache) getFromCache(docID, revID string, loadOnCacheMiss bo
 	return docRev, err
 }
 
+// In the event that a revision in invalid it needs to be replaced later and the revision cache value should not be
+// used. This function grabs the value directly from the bucket.
+func (rc *LRURevisionCache) LoadInvalidRevFromBackingStore(key IDAndRev, includeBody bool, includeDelta bool) (DocumentRevision, error) {
+	var delta *RevisionDelta
+	var docRevBody Body
+
+	value := revCacheValue{
+		key:     key,
+		invalid: true,
+	}
+	value.bodyBytes, value.body, value.history, value.channels, value.removed, value.attachments, value.deleted, value.expiry, value.err = revCacheLoader(rc.backingStore, key, includeBody)
```

Comment:
I had anticipated we could use revCacheLoaderForDoc here, at least when this is called from GetActive.  Is there a reason why we need to make the additional GetDocument call that's in revCacheLoader in all cases?

Reply:
Yeah there isn't a specific reason. I've updated this now. My solution is to modify `LoadInvalidRevFromBackingStore` to take a document. In the event a document is passed in which is the case under GetActive then it'll run revCacheLoaderForDoc. In the event that a document isn't passed in (a nil value) which is the case under `getFromCache` then it'll run `revCacheLoader`.
I think this is the best way to do it.